databricks/koalas/groupby.py

```diff
@@ -363,6 +363,207 @@ def any(self):
+           A     B  C
+        0  1   NaN  4
+        1  1   0.1  3
+        2  1  20.0  2
+        3  4  10.0  1
+
+        By default, iterates over rows and finds the sum in each column.
+
+        >>> df.groupby("A").cummax()
+              B  C
+        0   NaN  4
+        1   0.1  4
+        2  20.0  4
+        3  10.0  1
+
+        It works as below in Series.
+
+        >>> df.C.groupby(df.A).cummax()
+        0    4
+        1    4
+        2    4
+        3    1
+        Name: C, dtype: int64
+
+        """
+
+        return self._cum(F.max)
+
+    def cummin(self):
+        """
+        Cumulative min for each group.
+
+        Returns
+        -------
+        Series or DataFrame
+
+        See Also
+        --------
+        Series.cummin
+        DataFrame.cummin
+
+        Examples
+        --------
+        >>> df = ks.DataFrame(
+        ...     [[1, None, 4], [1, 0.1, 3], [1, 20.0, 2], [4, 10.0, 1]],
+        ...     columns=list('ABC'))
+        >>> df
+           A     B  C
+        0  1   NaN  4
+        1  1   0.1  3
+        2  1  20.0  2
+        3  4  10.0  1
+
+        By default, iterates over rows and finds the sum in each column.
+
+        >>> df.groupby("A").cummin()
+              B  C
+        0   NaN  4
+        1   0.1  3
+        2   0.1  2
+        3  10.0  1
+
+        It works as below in Series.
+
+        >>> df.B.groupby(df.A).cummin()
+        0     NaN
+        1     0.1
+        2     0.1
+        3    10.0
+        Name: B, dtype: float64
+        """
+        return self._cum(F.min)
+
+    def cumprod(self):
+        """
+        Cumulative product for each group.
+
+        Returns
+        -------
+        Series or DataFrame
+
+        See Also
+        --------
+        Series.cumprod
+        DataFrame.cumprod
+
+        Examples
+        --------
+        >>> df = ks.DataFrame(
+        ...     [[1, None, 4], [1, 0.1, 3], [1, 20.0, 2], [4, 10.0, 1]],
+        ...     columns=list('ABC'))
+        >>> df
+           A     B  C
+        0  1   NaN  4
+        1  1   0.1  3
+        2  1  20.0  2
+        3  4  10.0  1
+
+        By default, iterates over rows and finds the sum in each column.
+
+        >>> df.groupby("A").cumprod()
+              B     C
+        0   NaN   4.0
+        1   0.1  12.0
+        2   2.0  24.0
+        3  10.0   1.0
+
+        It works as below in Series.
+
+        >>> df.B.groupby(df.A).cumprod()
+        0     NaN
+        1     0.1
+        2     2.0
+        3    10.0
+        Name: B, dtype: float64
+
+        """
+        from pyspark.sql.functions import pandas_udf
+
+        def cumprod(scol):
+            # Note that this function will always actually called via `SeriesGroupBy._cum`,
+            # and `Series._cum`.
+            # In case of `DataFrameGroupBy`, it gose through `DataFrameGroupBy._cum`,
+            # `SeriesGroupBy.comprod`, `SeriesGroupBy._cum` and `Series._cum`
+            #
+            # This is a bit hacky. Maybe we should fix it.
```

Comment:
It's hacky ... but at least deduplicates all properly .. I will refactor this one alter when I happen to touch here again.